barretenberg/cpp/src/barretenberg/dsl/acir_format/witness_constant.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include "barretenberg/serialize/msgpack.hpp"
+#include "barretenberg/stdlib/primitives/field/field.hpp"
+
+namespace acir_format {
+template <typename FF> struct WitnessConstant {
```

Comment:
This name feels a little deceiving as we have a flag that states whether it is a constant. Perhaps something like  `WitnessOrConstant` or just `WitnessValue`?

Reply:
As this is what the name was before I will not block, it was just surprising to see `WitnessConstant` used in some places where they could possibly not be constants.

Reply:
that's a good point, I renamed to WitnessOrConstant.